README.md

```diff
@@ -1,6 +1,199 @@
+5. Iterate through the resource map and update the version of each.
+
+---
+
+### `bump --sync` or Propagation
+
+When resources are decoupled and stored in packages, there’s a need to preserve versions and maintain the current state of resources.
+`bump` works closely with `plasmactl compose`. The purpose of `compose` is to build the platform using resources from different places (domain repo, packages outside the repo).
+The purpose of `bump --sync` is to propagate the versions of a build to the current state.
+
+**How is it achieved?**
+
+#### Prerequisites:
+- It is important to use `bump --sync` after a fresh `compose`. Otherwise, an incorrect result or error may occur.
+
+### Overall Propagation Workflow:
+1. **Search and download the artifact to compare the build to:**
+  - Open the git repository.
+  - Search for the first bump commit after `HEAD`.
+  - Check if the artifact was previously downloaded and stored in the `.compose/artifacts` directory. If it exists, skip to step 5.
+  - Attempt to download the artifact with the name `repo_name-commit_hash_7_symbols-plasma-src.tar.gz` from the repository.
+  - Unarchive the artifact into the comparison directory (usually `.compose/comparison-artifact`).
+
+   > You can override the artifact commit with the `override` option, which bypasses the git history search and directly attempts to download the artifact.
+
+2. **Find the list of different files between the build and the artifact:**
+  - Compare the files in the build directory to the files in the artifact directory.
+  - If two files differ, their paths are added to the list of updated items.
+
+   **Excluded Subdirectories and Files:**
+  - `.git`
+  - `.compose`
+  - `.plasmactl`
+  - `.gitlab-ci.yml`
+  - `ansible_collections`
+  - `scripts/ci/.gitlab-ci.platform.yaml`
+  - `venv`
+
+   > @TODO: Include new files in the build directory but not in the artifact and note deleted files from the artifact.
+
+---
+
+### Propagating Versions
+
+Once you have the list of different files, filter out platform resources (to be covered in more detail). With the help of the inventory helper (which checks resources and variables), resources are divided into namespaces. These namespaces are split into **domain** and **packages**. While the domain is always static, packages come from `plasma-compose`.
+
+#### Syncing Resource Versions Across Namespaces:
+Next, we filter out resources that exist in multiple namespaces. `compose` handles this via merge strategies, but `sync` compares the build version with the namespace versions to select the correct resource. Once sorted, we iterate over the list of resources that differ between the build and the artifact.
+
+> @TODO: (Temporary) Should we include additional technical details here?
+
+### Iterating Through Resources:
+During this phase, the following rules apply:
+- If a resource is deleted from the build, it’s skipped. The developer should handle dependencies manually and bump related resources.
```

Comment:
I guess that could lead to some dependencies not being refreshed. 

What about propagating to dependencies using the commit sha of commit where resource was removed ?

Reply:
i don't think it's bump job to monitor if resource was deleted and re-check if it had dependencies which  developer forgot to remove/re-bump.
Can be done, but should be done after everything else.

Reply:
as we discussed and following your examples

```
resource X was removed and dependency to Y,Z was not removed by mistake > error
resource X was removed and dependency to Y,Z was removed as expected > propagate commit of X removal to Y,Z
-- how to know which resources used to depend on X since current state has changed > look in comparison artifact to see previous state
```
second case doesn't make sense, because at this point user should have removed dependencies and bumped their changes with `plasmactl bump`, which means we already have fresh version

First case can be handled in scope of `plasmactl bump --sync`